tools/lambda-promtail/lambda-promtail/kinesis.go

```diff
@@ -79,3 +79,39 @@ func ungzipData(data []byte) ([]byte, error) {
 
 	return io.ReadAll(reader)
 }
+
+func unmarshalData(data []byte) (events.CloudwatchLogsData, error) {
+	var recordData events.CloudwatchLogsData
+	err := json.Unmarshal(data, &recordData)
+	return recordData, err
+}
+
+func createLabels(record events.KinesisEventRecord, recordData events.CloudwatchLogsData) model.LabelSet {
+	labels := model.LabelSet{
+		model.LabelName("__aws_log_type"):                 model.LabelValue("kinesis"),
+		model.LabelName("__aws_kinesis_event_source_arn"): model.LabelValue(record.EventSourceArn),
+		model.LabelName("__aws_cloudwatch_log_group"):     model.LabelValue(recordData.LogGroup),
+		model.LabelName("__aws_cloudwatch_owner"):         model.LabelValue(recordData.Owner),
+	}
+
+	if keepStream {
+		labels[model.LabelName("__aws_cloudwatch_log_stream")] = model.LabelValue(recordData.LogStream)
+	}
```

Comment:
should these not be prefixed with `__aws_kinesis` instead of `__aws_cloudwatch`?

Reply:
@cstyan depending on the pov. The values are coming from cloudwatch, not from the K stream. K is a data stream not the source so I believe you want to have a filter that's a reflection of what your actual cloudwatch looks like, not the tool you used in the middle to stream that data.

Reply:
that makes sense, thanks for the context :+1: